mlflow/store/artifact/uc_volume_artifact_repo.py

```diff
@@ -0,0 +1,236 @@
+                ),
+                error_code=INVALID_PARAMETER_VALUE,
+            )
+
+        # The dbfs:/ path ultimately used for artifact operations should not contain the
+        # Databricks profile info, so strip it before setting `artifact_uri`.
+        super().__init__(remove_databricks_profile_info_from_artifact_uri(artifact_uri))
+
+        # Absolute path to the root of the volume. For example, '/Volumes/my-volume' for
+        # 'dbfs:/Volumes/my-volume'.
+        self.root_path = "/" + strip_scheme(self.artifact_uri).strip("/")
+        self.databricks_profile_uri = (
+            get_databricks_profile_uri_from_artifact_uri(artifact_uri)
+            or mlflow.tracking.get_tracking_uri()
+        )
+
+    def _relative_to_root(self, path):
+        return posixpath.relpath(path, self.root_path)
+
+    def _api_request(self, endpoint, method, **kwargs):
+        creds = get_databricks_host_creds(self.databricks_profile_uri)
+        return http_request(host_creds=creds, endpoint=endpoint, method=method, **kwargs)
+
+    def _list_directory_contents(self, directory_path: str, page_token: Optional[str] = None):
+        """
+        Lists the contents of a directory.
+
+        Args:
+            directory_path: The absolute path of a directory.
+
+        Returns:
+            The response from the API.
+
+        See also:
+            https://docs.databricks.com/api/workspace/files/listdirectorycontents
+        """
+        endpoint = f"{DIRECTORIES_API_ENDPOINT}{directory_path}"
+        return self._api_request(
+            endpoint=endpoint,
+            method="GET",
+            params={"page_token": page_token} if page_token else None,
+        )
+
+    def _paginated_list_directory_contents(
+        self, directory_path: str, page_token: Optional[str] = None
+    ):
+        response = self._list_directory_contents(directory_path, page_token)
+        if response.status_code == 404:
+            return []
+
+        augmented_raise_for_status(response)
+
+        response_json = response.json()
+        contents = response_json.get("contents", [])
+        if next_page_token := response_json.get("next_page_token"):
+            next_contents = self._paginated_list_directory_contents(directory_path, next_page_token)
+            return contents + next_contents
+        return contents
+
+    def _download(self, output_path: str, file_path: str):
+        """
+        Downloads a file.
+
+        Args:
+            output_path: The local path to save the downloaded file.
+            file_path: The absolute path of the file to download.
+
+        Returns:
+            The response from the API.
+
+        See also:
+            https://docs.databricks.com/api/workspace/files/download
+        """
+        endpoint = f"{FILES_API_ENDPOINT}{file_path}"
+        with open(output_path, "wb") as f:
+            with self._api_request(endpoint=endpoint, method="GET", stream=True) as resp:
```

Comment:
Do we need to disallow redirects when downloading?

Reply:
yea, thanks for the catch!

Reply:
In `mlflow/store/artifact/dbfs_artifact_repo.py`, redirects are only disallowed for `log_artifact` 🤔

https://github.com/mlflow/mlflow/blob/1533351efc56c68c79a39ee215a2747eef900855/mlflow/store/artifact/dbfs_artifact_repo.py#L119

Reply:
I suppose that makes sense for a data intercept security risk, which would be a primary attack vector on a compromised machine. SGTM